hnn_core/dipole.py

```diff
@@ -611,3 +677,59 @@ def write(self, fname):
         X = np.r_[X].T
 
         np.savetxt(fname, X, fmt=fmt, delimiter='\t')
+
+    def _write_hdf5(self, fname):
+        """Write dipole values to a hdf5 file.
+
+        Parameters
+        ----------
+        fname : str
+            Full path to the output file (.hdf5)
+        Outputs
+        -------
+        A hdf5 file containing the dipole object
+        """
+        print(f'Writing file {fname}')
+        dpl_data = dict()
+        dpl_data['times'] = self.times
+        dpl_data['sfreq'] = self.sfreq
+        dpl_data['nave'] = self.nave
+        dpl_data['data'] = self.data
+        dpl_data['scale_applied'] = self.scale_applied
+        write_hdf5(fname, dpl_data, overwrite=True)
```

Comment:
Should we be including a flag in the hdf5 file that specifies the type of HNN-core object that this should be used to create when read back in (i.e., to help prevent someone from accidentally trying to read in file that is incompatible with the `read_dipole()` method?

Reply:
Yes sure, I will add this.

Reply:
If you'd like to save this for later, that's fine too since I just noticed that this PR is already set to auto-merge!